packages/common/src/utils/filters.ts

```diff
@@ -1093,15 +1093,31 @@ export const reduceRequiredDimensionFiltersToFilterRules = (
     requiredFilters: MetricFilterRule[],
     table: Table,
     filters: FilterGroup | undefined,
+    explore: Explore,
 ): FilterRule[] =>
     requiredFilters.reduce<FilterRule[]>((acc, filter): FilterRule[] => {
         const filterRule = createFilterRuleFromRequiredMetricRule(
             filter,
             table.name,
         );
-        const dimension = Object.values(table.dimensions).find(
-            (tc) => getItemId(tc) === filterRule.target.fieldId,
-        );
+        const filterRefParts = filter.target.fieldRef.split('.');
```

Comment:
Looks a bit dangerous to me. Maybe we should add an assert function for field refs? If it passes the assert, it will execute the code; otherwise, it will log a warning to inform the user that the ref is incorrect.